### PR TITLE
Add a warning when NODE_SECURE_TOKEN is missing

### PR DESCRIPTION
**Pull request by tekeuange23**
I added a warning message on_commands("cwd | from | verify") by checking the environment variable NODE_SECURE_TOKEN

### CODE REVIEW

File: bin/index.js [hunk 1]
```diff
@@ -102,3 +111,9 @@ function defaultScannerCommand(name, options = {}) {
 
   return cmd;
 }
+
+function checkNodeSecureToken(){
+  if(process.env.NODE_SECURE_TOKEN){
+    kleur.yellow("Environment variable \"NODE_SECURE_TOKEN\" is missing: You need to login to npm with \`$ npm login\`")
```
Comment (fraxken) [inline bin/index.js:117]: How do you stdout the message if there is no `console.log` ?

File: bin/index.js [hunk 2]
```diff
@@ -102,3 +111,9 @@ function defaultScannerCommand(name, options = {}) {
 
   return cmd;
 }
+
+function checkNodeSecureToken(){
+  if(process.env.NODE_SECURE_TOKEN){
```
Comment (fraxken) [inline bin/index.js:116]: The condition is wrong, here you'r checking if the token is defined.

File: bin/index.js [hunk 3]
```diff
@@ -41,11 +41,17 @@ defaultScannerCommand("cwd", { strategy: vuln.strategies.NPM_AUDIT })
   .describe(i18n.getToken("cli.commands.cwd.desc"))
   .option("-n, --nolock", i18n.getToken("cli.commands.cwd.option_nolock"), false)
   .option("-f, --full", i18n.getToken("cli.commands.cwd.option_full"), false)
-  .action(commands.scanner.cwd);
+  .action((_, _, opts) => {
+    checkNodeSecureToken();
+    commands.scanner.cwd(opts);
```
Comment (fraxken) [inline bin/index.js:46]: What if the Async function throw an Error ?

**Review by fraxken — CHANGES_REQUESTED** [review]

**Comment by fraxken** [comment]
You should test and run your own implementation by the way

**Comment by tekeuange23** [comment]
It should be good now!

File: bin/index.js [hunk 4]
```diff
@@ -102,3 +127,18 @@ function defaultScannerCommand(name, options = {}) {
...
+function checkNodeSecureToken() {
+  if (!process.env.NODE_SECURE_TOKEN) {
+    console.log(
+      kleur
+        .white()
+        .bold(
+          `\n ${kleur
+            .yellow()
+            .bold(`Environment variable ${"\"NODE_SECURE_TOKEN\""} is missing.
+            \trun${"`npm login`"} to login`)}`
```
Comment (fraxken) [inline bin/index.js:140]: Because there is no relation with npm login you need to remove the last part of the log

File: bin/index.js [hunk 5]
```diff
@@ -6,6 +6,7 @@ dotenv.config();
 import { createRequire } from "module";
 
 // Import Third-party Dependencies
+import { execSync } from "node:child_process";
```
Comment (fraxken) [inline bin/index.js:9]: Remove this import ?

**Review by fraxken — CHANGES_REQUESTED** [review]

File: bin/index.js (hunk 4 above)
Comment (tekeuange23) [reply bin/index.js:140]: Sure

File: test/commands/summary.spec.js [hunk 6]
```diff
@@ -46,3 +50,11 @@ test("summary should execute summary command on fixtures 'result-test1.json'", a
...
   tape.end();
 });
+
+test("warning on missing 'NODE_SECURE_TOKEN'", async(tape) => {
+  if (!process.env.NODE_SECURE_TOKEN) {
+    tape.equal(process.env.NODE_SECURE_TOKEN, undefined);
+  }
+
+  tape.end();
+});
```
[the remark this answers is not in the document]
Comment (tekeuange23) [reply test/commands/summary.spec.js:60]: I mean I have to remove the If statement due to the test name 🤔

File: bin/index.js (hunk 5 above)
Comment (tekeuange23) [reply bin/index.js:9]: Sure useless ^^

**Pull request merged**